### DISCUSSION TITLE
Graphql api draft option not working for unpublished documents

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#1569] [Q&A] [answered]
# Bug Report

I found our draft articles and unpublished articles are shown in prod env.

## Steps to Reproduce

1. Open pages list in demo site and publish the first article
2. Publish the second article and then unpublish it directly
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/5801015/205217465-2d7cc1bc-f084-45f9-abe5-25fde5cfbb19.png">

3. Open the graphql-playground: 
https://demo.payloadcms.com/api/graphql-playground

Input this query and execute:

```
# Write your query or mutation here
query getPages {
  Pages(draft: false) {
    docs {
      slug
      _status
      updatedAt
    }
  }
}
```

<img width="1511" alt="image" src="https://user-images.githubusercontent.com/5801015/205217598-ecc34f49-dfe8-4c71-8675-6184de63a07d.png">

**Comment by Stupidism** [comment]
Do I have to use this query to filter out `_status: "draft"`? If so, can I find any documentation about this?

```
# Write your query or mutation here
query getPages {
  Pages(draft: false, where:{_status: {equals: published}}) {
    docs {
      title
      slug
      _status
      updatedAt
    }
  }
}
```

<img width="1569" alt="image" src="https://user-images.githubusercontent.com/5801015/205217888-53504f23-1845-4307-9f99-a4f8637b14f7.png">

**Comment by jmikrut** [comment, marked as answer]
Your understanding in your second comment is correct - the `draft: true` argument _**replaces any docs in your main collection with their newest drafts, which would be stored in the `_versions`**_ collection.

We do have docs about this - take a look here:
https://payloadcms.com/docs/versions/drafts#controlling-who-can-see-collection-drafts

Basically, the `draft` argument alone does _**not**_ restrict who can see drafts. If you need that, you need to do that with access control. The `draft` arg simply does what is stated above.

Does that make sense? I will convert this to a discussion as I don't believe there's anything actionable to be done here. What do you think?

**Reply by Stupidism** [reply]
Thanks, i didn't understand this doc correctly.

**Comment by Stupidism** [comment]
@jmikrut An additional question, as I checked the doc and the ts interface, it seems there's no way I can get the graphql query variables from the access function.

I want to implement sth like this as the default access for our articles:


```
  access: {
    read: ({ req }) => {
      console.log('req: ', req.params);
      
      // to get the draft option from request
      if (req.params.draft) return true;

      return {
        _status: {
          equals: 'published',
        },
      };
    },
  },
```
